src/clr-angular/utils/popover/popover-content.ts

```diff
@@ -98,25 +111,48 @@ export class ClrPopoverContent implements AfterContentChecked, OnDestroy {
     // Create the view container
     this.view = this.container.createEmbeddedView(this.template);
     this.smartEventsService.contentRef = this.view.rootNodes[0]; // So we know where/what to set close focus on
-    // Position the content and add a click listener
     this.renderer.addClass(this.view.rootNodes[0], 'clr-popover-content');
+    // Reset to the begining of the document to be available for sizing/positioning calculations.
+    // If we add new content to the bottom it triggers changes in the layout that may lead to false anchor
+    // coordinates values.
+    this.renderer.setStyle(this.view.rootNodes[0], 'top', '0px');
```

Comment:
Why are we changing to the top/left positioning from what we had before?

Reply:
In order to make the calculations we need to have the element attached to the DOM (sounds unnecessary, but Matt can confirm that in practice we need it). It used to be added to the bottom of the document and after the calculations are done, it gets relocated to the proper position on the body. This works in the static case, but in the dynamic case it changes the overall page size and this affects negatively the smart position relocation logic.
So, as described in the comment above it, I am initializing it at absolute 0/0 instead, which detaches it from the inline DOM structure but still keeps it in the DOM. This way the reposition coordinates calculation is okay.
The actual negative effect was that the repositioned popover was moved to the top with as many pixels as its height was.

Reply:
👍  Thanks!